### PR TITLE
Fix path to server when running platform neutral vsix

### PR DESCRIPTION
**Pull request by dibarbet**
Won't fully work until https://github.com/dotnet/roslyn/pull/69110 is in

### CODE REVIEW

File: src/lsptoolshost/roslynLanguageServer.ts [hunk 1]
```diff
@@ -659,8 +657,10 @@ function getServerPath(options: Options, platformInfo: PlatformInformation) {
     return serverPath;
 }
 
-function getServerFileName(platformInfo: PlatformInformation) {
-    const serverFileName = 'Microsoft.CodeAnalysis.LanguageServer';
+function getInstalledServerPath(platformInfo: PlatformInformation) {
```
Comment (jasonmalinowski) [inline src/lsptoolshost/roslynLanguageServer.ts:660]: Are we moving to always stating return types for functions or not?

File: src/lsptoolshost/roslynLanguageServer.ts [hunk 2]
```diff
@@ -659,8 +657,10 @@ function getServerPath(options: Options, platformInfo: PlatformInformation) {
...
+    const clientRoot = __dirname;
+    const serverFilePath = path.join(clientRoot, '..', '.roslyn', 'Microsoft.CodeAnalysis.LanguageServer');
+
     let extension = '';
     if (platformInfo.isWindows()) {
```
Comment (jasonmalinowski) [inline src/lsptoolshost/roslynLanguageServer.ts:665]: Not related to your change but it bugs me the isMacOS isn't an else-if....

File: src/lsptoolshost/roslynLanguageServer.ts [hunk 3]
```diff
@@ -673,7 +673,13 @@ function getServerFileName(platformInfo: PlatformInformation) {
         extension = '.dll';
     }
 
-    return `${serverFileName}${extension}`;
+    let pathWithExtension = `${serverFilePath}${extension}`;
+    if (!fs.existsSync(pathWithExtension)) {
```
Comment (jasonmalinowski) [inline src/lsptoolshost/roslynLanguageServer.ts:677]: So on Linux we don't set extension, expect this to fail and then take the if block?

**Review by jasonmalinowski — APPROVED** [review]

File: src/lsptoolshost/roslynLanguageServer.ts (hunk 1 above)
Comment (dibarbet) [reply src/lsptoolshost/roslynLanguageServer.ts:660]: we should, but I haven't enabled the rule, hence me making bad changes :) ill fix

File: src/lsptoolshost/roslynLanguageServer.ts (hunk 3 above)
Comment (jasonmalinowski) [reply src/lsptoolshost/roslynLanguageServer.ts:675]: Oh right, I forgot there'd be no extension. It _might_ make sense to put a comment there that the lack of setting that is intentional, but...

**Review by jasonmalinowski — APPROVED** [review]

**Pull request merged**